lib/icedfrisby.js

```diff
@@ -1073,33 +1073,136 @@ Frisby.prototype.setResponseHeader = function(key, value) {
+      mochaContext.results_.failedCount = 0;
+
+      process.stdout.write('R');
+      self._makeRequest(mochaContext, done);
+    } else {
+      // should abort instead (it.spec.fail ?)
+      mochaContext.results_.failedCount = 1;
+      // In frisby it.results_ would trigger a failure for jasmine but has
+      // no effect in mocha. We need to indicate a failure for tests that
+      // reach this point.
+      const err ='Destination URL may be down or URL is invalid';
+      self._finish(err, done);
+      // assert();
+    }
+  }, self._timeout);
+
+  self.current.it(function (data) {
+    if (!timeoutFinished) {
+      clearTimeout(timeoutId);
+      self._performInspections.call(self);
+      self._invokeExpects.call(self, mochaContext, done);
+    }
+  });
+};
+
+
+// called from makeRequest if request has finished successfully
+Frisby.prototype._invokeExpects = function (mochaContext, done) {
+  mochaContext.response = this.current.response;
+  this.current.expectsFailed = true;
+
+  // if you have no expects, they can't fail
+  if (this.current.expects.length === 0) {
+    this.current.expectsFailed = false;
+  }
+
+  // REQUIRES count for EACH loop iteration (i.e. DO NOT OPTIMIZE THIS LOOP)
+  // Some 'expects' helpers add more tests when executed (recursive 'expectJSON' and 'expectJSONTypes', with nested JSON syntax etc.)
+  for (let i = 0; i < this.current.expects.length; i++) {
+    if (false !== this._exceptionHandler) {
+      try {
+        this.current.expects[i].call(it);
+      } catch (e) {
+        this._exceptionHandler.call(this, e);
+      }
+    } else {
+      this.current.expects[i].call(it);
+    }
+  }
+
+  if (mochaContext.results_.failedCount === 0) {
+    this.current.expectsFailed = false;
+  }
+
+  this._finish(null, done);
+};
+
+
+// Execute further expects for the current spec (called from _invokeExpects)
+Frisby.prototype._finish = function (err, done) {
+  if (this.current.after) {
+    if (this.current.expectsFailed && this.current.outgoing.inspectOnFailure) {
+      console.log(this.current.itInfo +
+        ' has FAILED with the following response:');
+      this.inspectStatus();
+      this.inspectJSON();
+    }
+
+    // REQUIRES count for EACH loop iteration (i.e. DO NOT OPTIMIZE THIS LOOP)
+    // this enables after to add more after to do things (like inspectJSON)
+    for (let i = 0; i < this.current.after.length; i++) {
+      const fn = this.current.after[i];
+      if (false !== this._exceptionHandler) {
+        try {
+          fn.call(this);
+        } catch (e) {
+          this._exceptionHandler(e);
+        }
+      } else {
+        fn.call(this);
+      }
+    }
+  }
+
+  // finally call done to finish spec
+  if (err) {
+    done(new Error(err));
+  } else {
+    done();
+  }
+};
+
+
 /**
- * @param {number} retry - n retries
  * @return {object}
  * @desc Run the current Frisby test
  */
-Frisby.prototype.toss = function (retry) {
```

Comment:
At present, the `retry` parameter is broken. Does anyone know how recently it worked? Trying to judge whether or not it's a breaking change. If it's _never_ worked for example, then removing it would certainly not be a breaking change.